frontend/src/components/smart-contracts/hooks/useSmartContractCall.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState, useCallback } from 'react';
+
+type CallContractCallback = () => Promise<any>;
+
+export const useSmartContractCallback = <ResultType>(
```

Comment:
I don't really like this hook. It's hard to tell what it's doing. Can't we just add this to `job` and do

```
const {job} = useJob();
if job.timeout > date.now() etc...
```

IMHO - this is stuff that should be abstracted behind our existing hooks so its very easy to use at component level.

Reply:
It is not a call on a specific job.  It is a call to the Job smart contract.

I was trying to make a general purpose hook.  But I think what you are looking for is a more specific hook like:

```typescript
const [jobTimeoutSeconds, timeoutLoading] = useJobContractVariable('COMPLETED_TIMEOUT_SECONDS');
```

Reply:
I created a `useJobContractVariable` hook.  Is there a better way?

Reply:
I refactored this to `useSmartContractCall`.  I think this is the cleanest way to do this.

See https://github.com/engineer-dao/platform/pull/136/commits/4f21a31020e4e88ffa29792b843ed1dea985e02f#diff-1d48d830e080e7d6c77b002a29f0025f0f8b86a4da4a00f795782439ec9d1cd2R16-R19